lib/yaml/constructor.py

```diff
@@ -495,6 +503,12 @@ def construct_undefined(self, node):
         SafeConstructor.construct_undefined)
 
 class FullConstructor(SafeConstructor):
+    # 'extend' is blacklisted because it is used by
+    # construct_python_object_apply to add `listitems` to a newly generate
+    # python instance
+    STATE_BLACKLIST_KEYS = ['^extend$', '^__.*__$']
```

Comment:
Maybe `BLACKLIST_STATE_KEYS` is more precise. `STATE` is there because these are the blacklisted keys for the object state (indeed it is used by `set_python_instance_state`)

Reply:
Ok, added a test. Hope it's good where I placed it.

Reply:
Thanks for the explanation. Per my last review let's change it to `STATE_KEYS_BLACKLIST` and make it user extendable with a `state_keys_blacklist` list of regexps.